rust/chains/hyperlane-ethereum/src/fallback.rs

```diff
@@ -97,30 +100,36 @@ impl JsonRpcClient for FallbackProvider<PrometheusJsonRpcClient<Http>> {
         let params = serde_json::to_value(params).expect("valid");
 
         let mut errors = vec![];
-        for (idx, provider) in self.0.iter().enumerate() {
-            let fut = match params {
-                Value::Null => provider.request(method, ()),
-                _ => provider.request(method, &params),
-            };
-
-            match fut.await {
-                Ok(v) => return Ok(serde_json::from_value(v)?),
-
-                Err(HttpClientError::ReqwestError(e)) => {
-                    warn!(error=%e, provider_index=%idx, ?provider, method, "ReqwestError in http provider; falling back to the next provider");
-                    errors.push(HttpClientError::ReqwestError(e).into())
-                }
-                Err(HttpClientError::SerdeJson { err, text }) => {
-                    warn!(error=%err, text, provider_index=%idx, ?provider, method, "ReqwestError in http provider; falling back to the next provider");
-                    errors.push(HttpClientError::SerdeJson { err, text }.into())
-                }
-                Err(HttpClientError::JsonRpcError(e)) => {
-                    if METHODS_TO_NOT_TO_FALLBACK_ON.contains(&method) {
-                        warn!(error = %e, provider_index=%idx, ?provider, method, "JsonRpcError in http provider; not falling back");
-                        return Err(HttpClientError::JsonRpcError(e).into());
-                    } else {
-                        warn!(error = %e, provider_index=%idx, ?provider, method, "JsonRpcError in http provider; falling back to the next provider");
-                        errors.push(HttpClientError::JsonRpcError(e).into())
+        // make sure we do at least 4 total retries.
```

Comment:
One of the chains kept crashing because it only had one url specified (alfajores). I was using our production configuration so this would happen out in the wild too. Since we deploy the scraper to scrape ALL chains and not just one, this means that it completely broke. Since I assume we want to keep using fallback provider, this seemed like a reasonable approach to increasing the redundancy when we only list one URL.

Reply:
cool, yeah I'm up for the change